### DISCUSSION TITLE
Budibase Okta SSO integration error

### DISCUSSION THREAD

**Discussion by ryanheinlein-tcp** [Budibase/budibase#3510] [Help]
I followed the instructions here https://docs.budibase.com/building-apps/admin-and-management/authentication/oidc as well as here https://docs.budibase.com/building-apps/admin-and-management/authentication/okta. 

When trying to sign in via okta I get the error in the picture
<img width="908" alt="Screen Shot 2021-11-23 at 2 48 50 PM" src="https://user-images.githubusercontent.com/79862685/143093664-b4d60c3e-57b8-409d-aa2a-73701762ea65.png">

**Comment by Rory-Powell** [comment]
Hey @ryanheinlein-tcp 👋 

It's hard to tell anything specific based on the screenshot alone. 

Could you include some more details by showing us 
- The okta URL that budibase redirected you to when you clicked sign in 
- The browser dev tools response for the failing 400 request
- The redirect url you have configured in okta for budibase

Hopefully there will be enough details in one of these to get an insight into what is going wrong

**Reply by ryanheinlein-tcp** [reply]
https://thecommonsproject.okta.com/.well-known/openid-configuration is the config url in budibase.

My sign-in redirect URL in okta is: https://budibase.tcpdev.org/api/admin/auth/oidc/callback

The error I am seeing in dev tools on chrome is this:  The 'redirect_uri' parameter must be a Login redirect URI in the client app settings

Please let me know if you need anything else

**Reply by mjashanks** [reply]
@ryanheinlein-tcp - check your `Platform URL` in organisation. It should be set to `https://budibase.tcpdev.org`

![image](https://user-images.githubusercontent.com/3524181/143915117-f51417b3-4c8e-4d92-8a29-001959ec2d10.png)

**Reply by ryanheinlein-tcp** [reply]
It is set to that

**Reply by Rory-Powell** [reply]
Thank you for the details, it looks like it's the path causing bother. When redirecting to an SSO provider budibase sends a request to our own api to get the details for the provider, i.e. the url to authenticate with, which is enriched with elements from the config e.g. client id / secret

The callback URL is generated at this time:
- use `https` 
- use the incoming hostname of the request e.g. `budibase.tcpdev.org`
- use a predefined path `/api/global/auth/oidc/callback`

It looks like your path is set to `api/admin/auth/oidc/callback`, the key part here is the `admin` instead of `global`. Previously `admin` was used for the google authentication and can still be used for backwards compatibility. OIDC has always used `global` afaik, is your UI showing else in the placeholder?

**Reply by Rory-Powell** [reply]
It's worth mentioning and mike has already touched on it, we're changing the behaviour of generating the callback url as it's non-deterministic, in the next release (this week) we are pushing a change to synchronize the callback url with the configured platform url. This mean that the now complete callback url will be shown on the UI
